jsapp/js/components/usageLimits/overLimitBanner.component.tsx

```diff
@@ -13,40 +20,166 @@ interface OverLimitBannerProps {
+};
+
+/*
+ * Returns a formatted string with all the limits
+ */
+const getAllLimits = (limits: string[], interval: string) => {
+  if (limits.length === 0) {
+    return '';
+  }
+
+  const items = limits.map((limit) => `**${getLimitText(limit, interval)}**`);
+
+  if (items.length === 1) {
+    return `${items[0]}`;
+  }
+
+  return `${items.slice(0, -1).join(', ')} and ${items.slice(-1)[0]}`;
+};
+
+/*
+ * Returns the base message to be displayed in the banner
+ */
+const getMessage = ({
+  isWarning,
+  isMmo,
+  isTeamLabelActive,
+  limits,
+}: {
+  isWarning: boolean;
+  isMmo: boolean;
+  isTeamLabelActive: boolean;
+  limits: string;
+}) => {
+  let message = '';
+  if (isWarning) {
+    if (isMmo && isTeamLabelActive) {
+      message = t('Your team is approaching the ##LIMITS## limit.');
+    } else if (isMmo) {
+      message = t('Your organization is approaching the ##LIMITS## limit.');
+    } else {
+      message = t('You are approaching the ##LIMITS## limit.');
+    }
+  } else if (isMmo && isTeamLabelActive) {
+    message = t('Your team has reached the ##LIMITS## limit.');
+  } else if (isMmo) {
+    message = t('Your organization has reached the ##LIMITS## limit.');
+  } else {
+    message = t('You have reached the ##LIMITS## limit.');
+  }
+
+  return message.replace('##LIMITS##', limits);
+};
+
 const OverLimitBanner = (props: OverLimitBannerProps) => {
   const navigate = useNavigate();
-  if (!props.limits.length) {
+
+  const orgQuery = useOrganizationQuery();
+
+  if (
+    !orgQuery.data ||
+    !envStore.isReady ||
+    !subscriptionStore.isInitialised ||
+    !props.limits.length
+  ) {
+    return null;
+  }
+
+  const {limits, interval, warning} = props;
+  const {is_mmo: isMmo, request_user_role: userRole} = orgQuery.data;
+  const subscription = subscriptionStore.activeSubscriptions[0];
+
+  // Get all the limits in a formatted string
+  const allLimits = getAllLimits(limits, interval);
+
+  // If the user is a member of an MMO, we don't show a warning:
+  if (isMmo && userRole === OrganizationUserRole.member && !!warning) {
     return null;
   }
+
+  // Get the first part of the message
+  const message1 = getMessage({
+    isWarning: !!warning,
+    isMmo,
+    isTeamLabelActive: shouldUseTeamLabel(envStore.data, subscription),
+    limits: allLimits,
+  });
+
+  // We have different second part of the message for admins and owners of MMOs
+  let message2 = '';
+  if (isMmo) {
+    if (userRole === OrganizationUserRole.owner) {
+      if (warning) {
+        message2 = t('Purchase additional submissions add-ons to continue collecting and submitting data.');
+      } else {
+        message2 = t('Please purchase an add-on to increase your submission limits.');
+      }
+    } else if (warning) {
+      message2 = t("Once the limit has been reached, you won't be able to collect or submit any new data until the team owner has purchased additional submissions.");
```

Comment:
'Team' shouldn't be hardcoded here. This text is already quite complex, though. I will leave it up to you whether you want to add in more logic or just cut out the term and say "until the owner".